test/mpi/util/mtest_thread_abt.h

```diff
@@ -229,3 +247,39 @@ void MTest_finalize_thread_pkg()
     ret = ABT_finalize();
     MTEST_ABT_ERROR(ret, "ABT_finalize");
 }
+
+char MTest_argobots_lock = 0;   /* Atomic */
+int MTest_argobots_refcount = 0;        /* Protected by MTest_argobots_lock */
+
+static void MTest_abt_acquire_spinlock(void)
+{
+    /* TODO: support platforms that do not support GCC's __atomic builtins. */
+    while (!__atomic_test_and_set(&MTest_argobots_lock, __ATOMIC_ACQ_REL));
+}
+
+static void MTest_abt_release_spinlock(void)
+{
+    __atomic_clear(&MTest_argobots_lock, __ATOMIC_RELEASE);
+}
+
+static void MTest_abt_incr_refcount(void)
+{
+    MTest_abt_acquire_spinlock();
```

Comment:
Will it work without the lock?

Reply:
Lock initialization might happen in created threads, so this spinlock is necessary.

Reply:
I think this is beyond simple. It adds the lazy initialization logic, then the concern of checking and initialization inside the spawned threads, then the concern of using spin-lock, and then the concern of atomics. I much prefer solution #2 at this point.

Reply:
Thank you. I got it. I will adopt solution #2.

Reply:
This has been fixed by #4429, so this PR no longer contains it.